SmartDeviceLinkTests/DevAPISpecs/SDLFileManagerSpec.m

```diff
@@ -256,6 +258,142 @@ @implementation FileManagerSpecHelper
         });
     });
 
+    describe(@"check hasUploadedFile response", ^{
+        __block SDLFile *testFile = nil;
+
+        context(@"on RPC version >= 4.4.0", ^{
+            beforeEach(^{
+                [SDLGlobals sharedGlobals].rpcVersion = [SDLVersion versionWithMajor:7 minor:1 patch:0];
+                [testFileManager.stateMachine setToState:SDLFileManagerStateReady fromOldState:SDLFileManagerStateShutdown callEnterTransition:NO];
+
+                NSData *testFileData = [@"someData" dataUsingEncoding:NSUTF8StringEncoding];
+                testFile = [SDLFile persistentFileWithData:testFileData name:@"testFile4" fileExtension:@"png"];
+            });
+
+            context(@"when the file is in remoteFileNames", ^{
+                beforeEach(^{
+                    testFileManager.mutableRemoteFileNames = [NSMutableSet setWithArray:testInitialFileNames2];
+                });
+
+                it(@"should return NO", ^{
+                    expect([testFileManager hasUploadedFile:testFile]).to(equal(YES));
+                });
+            });
+
+            context(@"when the file is not in remoteFileNames", ^{
+                beforeEach(^{
+                    testFileManager.mutableRemoteFileNames = [NSMutableSet setWithArray:testInitialFileNames];
+                });
+
+                it(@"should return NO", ^{
+                    expect([testFileManager hasUploadedFile:testFile]).to(equal(NO));
+                });
+            });
+        });
+
+        context(@"on RPC version < 4.4.0", ^{
+            beforeEach(^{
+                [SDLGlobals sharedGlobals].rpcVersion = [SDLVersion versionWithMajor:4 minor:3 patch:0];
+                [testFileManager.stateMachine setToState:SDLFileManagerStateReady fromOldState:SDLFileManagerStateShutdown callEnterTransition:NO];
+            });
+
+            context(@"when the file is persistent", ^{
+                beforeEach(^{
+                    NSData *testFileData = [@"someData" dataUsingEncoding:NSUTF8StringEncoding];
+                    testFile = [SDLFile persistentFileWithData:testFileData name:@"testFile4" fileExtension:@"png"];
+                });
+
+                it(@"should return NO", ^{
+                    expect([testFileManager hasUploadedFile:testFile]).to(equal(NO));
+                });
```

Comment:
Isn't this identical to the below `context(@"when the file is not in remoteFileNames", ^{`